src/main/java/com/questhelper/skills/agility/Agility.java

```diff
@@ -44,58 +44,86 @@
 import java.util.List;
 import net.runelite.api.ItemID;
 import net.runelite.api.ObjectID;
-import net.runelite.api.Skill;
+import static net.runelite.api.Skill.AGILITY;
 import net.runelite.api.coords.WorldPoint;
 
 @QuestDescriptor(
 	quest = QuestHelperQuest.AGILITY
 )
 public class Agility extends ComplexStateQuestHelper
-{
 
+	// TODO::Rellekka and Ardougne
+{
 	// Items recommended
 	ItemRequirement bootsOfLightness, gracefulOutfit, gracefulHood, gracefulTop, gracefulLegs, gracefulGloves, gracefulBoots, gracefulCape;
 
 	SkillRequirement ag10, ag20, ag30, ag40, ag50, ag60, ag70, ag80, ag90;
 
 	SkillRequirement ag45;
 
-	QuestStep gnomeStronghold, draynorVillage, alKharid, varrock, canifis, falador, seersVillage, pollnivneach, rellekka, ardougne;
+	AgilityCourse gnomeStronghold, draynorVillage, alKharid, varrock, canifis, falador, seersVillage, pollnivneach;
+	ConditionalStep gnomeStep, draynorStep, alKharidStep, varrockStep, canifisStep, faladorStep, seersStep, pollnivneachStep;
+	QuestStep rellekka, ardougne;
 
 	@Override
 	public QuestStep loadStep()
 	{
+		setupCourses();
```

Comment:
With the All Items Helper, it runs 'setupRequirements' for each helper, and needs to have everything instantiated within it. To ensure the courses are setup as they're used for the reqs, moving this setup into setupRequirement should work fine!